sys/dev/pci/pci_user.c

```diff
@@ -986,7 +1000,7 @@ pci_bar_mmap(device_t pcidev, struct pci_bar_mmap *pbm)
 	flags = MAP_SHARED;
 	addr = 0;
 	if ((pbm->pbm_flags & PCIIO_BAR_MMAP_FIXED) != 0) {
-		addr = (uintptr_t)pbm->pbm_map_base;
+		addr = (__cheri_addr vm_ptr_t)pbm->pbm_map_base;
 		flags |= MAP_FIXED;
```

Comment:
Hmm, this code needs the whole mmap addr treatment (MAP_FIXED requires a VMMAP bearing cap or a NULL-derived cap which implies MAP_EXCLUSIVE).

Reply:
Should be fixed now